cpp/DataStorm/clock/CMakeLists.txt

```diff
@@ -5,15 +5,19 @@ project(datastorm_clock CXX)
 include(../../cmake/common.cmake)
 
 add_executable(reader Reader.cpp)
-target_link_libraries(reader Ice::Ice Ice::DataStorm)
+target_link_libraries(reader PRIVATE Ice::Ice Ice::DataStorm)
```

Comment:
What does PRIVATE means here?

Reply:
It means link with the library but don't include it int he libraries link list. Like bzip2 in Ice. For executables it doesn't really matter you put there as nothing links with an executable. It's more signaling intent. 

For a library it's really about if the library exposes APIs from its dependencies. https://leimao.github.io/blog/CMake-Public-Private-Interface/